### PR TITLE
doc: add Trusty ACRN doc

### PR DESCRIPTION
**Pull request by dbkinder**
Add the Trusty and Security Services in ACRN tutorial

Signed-off-by: David B. Kinder <david.b.kinder@intel.com>

### CODE REVIEW

**Comment by dbkinder** [comment]
Merged now so we'll have this in the 0.6 release tag (we can submit important fixes after the tag if required)

File: doc/tutorials/trustyACRN.rst [hunk 1]
```diff
@@ -0,0 +1,408 @@
...
+Trusty in ACRN
+**************
+
+ACRN is a flexible, lightweight reference hypervisor, built with
+real-time and safety-criticality in mind, optimized to streamline
+embedded development through an open source platform. In this
+section, we'll focus on two major components:
+
+* one is the basic idea of
+  secure world and unsecure world isolation (so called one-vm,
```
Comment (YadongQi) [inline doc/tutorials/trustyACRN.rst]: Normally, we use "non-secure".

File: doc/tutorials/trustyACRN.rst [hunk 2]
```diff
@@ -0,0 +1,408 @@
...
+
+Using Android as an example, Keystore functions are explained in greater
+details in this `Android keymaster functions document
+<https://source.android.com/security/keystore/implementer-ref>`_
+
+.. figure:: images/trustyacrn-image3.png
+   :align: center
+   :width: 600px
+   :name: keymaster-app
+
```
Comment (YadongQi) [inline doc/tutorials/trustyACRN.rst]: Could we also have a title for this figure like trusty-arch? We can name this "Keymaster Functions"

File: doc/tutorials/trustyACRN.rst [hunk 3]
```diff
@@ -0,0 +1,408 @@
...
+is controlled by normal world driver, Trusty needs to send an RPMB data
+frame ( encrypted by hardware-backed unique encryption key and signed by
+AuthKey) over Trusty IPC channel to Trusty SS proxy daemon, which then
+forwards RPMB data frame to physical RPMB partition in eMMC.
+
+.. figure:: images/trustyacrn-image2.png
+   :align: center
+   :width: 600px
+   :name: trusty-ss-ta
+
```
Comment (YadongQi) [inline doc/tutorials/trustyACRN.rst]: title for this figure: "Secure Storage Architecture"

File: doc/tutorials/trustyACRN.rst [hunk 4]
```diff
@@ -0,0 +1,408 @@
...
+client App in normal world and a Trusted App (TA) in Trusty OS. The OEM
+TA then can talk with either TD or TP (or both) of SS TA through Trusty
+internal process IPC to request TA-specific secure file
+open/creation/deletion/read/write operations.
+
+.. figure:: images/trustyacrn-image5.png
+   :align: center
+   :width: 600px
+   :name: trusty-ss-ta-storage
+
```
Comment (YadongQi) [inline doc/tutorials/trustyACRN.rst]: title: "Develop TA/CA using SS"

File: doc/tutorials/trustyACRN.rst [hunk 5]
```diff
@@ -0,0 +1,408 @@
...
+As previously mentioned, Trusty Secure Monitor could be any
+hypervisor. In the ACRN project the ACRN hypervisor will behave as the
+secure monitor to schedule in/out Trusty secure world.
+
+.. figure:: images/trustyacrn-image4.png
+   :align: center
+   :width: 600px
+   :name: trusty-isolated
+
```
Comment (YadongQi) [inline doc/tutorials/trustyACRN.rst]: title: "One-VM, Two-Worlds Architecture"

File: doc/tutorials/trustyACRN.rst [hunk 6]
```diff
@@ -0,0 +1,408 @@
...
+platform.
+
+:numref:`trusty-rpmb` shows an overview of the virtualization of secure storage
+high-level architecture.
+
+.. figure:: images/trustyacrn-image6.png
+   :align: center
+   :width: 600px
+   :name: trusty-rpmb
+
```
Comment (YadongQi) [inline doc/tutorials/trustyACRN.rst]: title: "RPMB Virtualization"

**Comment by YadongQi** [comment]
Hi, @dbkinder I reviewed: https://projectacrn.github.io/latest/tutorials/trustyACRN.html#, some comments in the commit. Could you help update them later?
@bzhu5, For the figure name, could you help double check them?

File: doc/tutorials/trustyACRN.rst (hunk 2 above)
Comment (yang8621) [reply doc/tutorials/trustyACRN.rst]: Agree. seems Figure XXX is missing if no title.

File: doc/tutorials/trustyACRN.rst [hunk 7]
```diff
@@ -0,0 +1,408 @@
...
+Trusty implements a secure storage services (in Secure Storage TA) based
+on RPMB (Replay Protected Memory Block) partition in eMMC or UFS flash
+storage. The details of how RPMB works are out of scope in this article.
+You can read the `eMMC/UFS JEDEC specification
+<https://www.jedec.org/standards-documents/focus/flash/universal-flash-storage-ufs>`_
+to understand that.
+
+This secure storage can provide data confidentiality, integrity, and
+anti-replay protection.Confidentiality is guaranteed by data encryption
```
Comment (yang8621) [inline doc/tutorials/trustyACRN.rst]: space

File: doc/tutorials/trustyACRN.rst [hunk 8]
```diff
@@ -0,0 +1,408 @@
...
+with a root key derived from the platform chipset's unique key/secret.
```
Comment (yang8621) [inline doc/tutorials/trustyACRN.rst]: root key -> encryption key, to align with line 185: 
"encrypted by hardware-backed unique encryption key"

File: doc/tutorials/trustyACRN.rst [hunk 9]
```diff
@@ -0,0 +1,408 @@
...
+key).  See `Android Key and ID Attestation
+<https://source.android.com/security/keystore/attestation>`_
+for details.
+
+In Trusty, the secure storage architecture is shown in the figure below.
+In the secure world, there is a SS (Secure Storage) TA, which has an
+RPMB authentication key (AuthKey, an HMAC key) and uses this Authkey to
+talk with the RPMB controller in the eMMC device. Since the eMMC device
+is controlled by normal world driver, Trusty needs to send an RPMB data
+frame ( encrypted by hardware-backed unique encryption key and signed by
```
Comment (yang8621) [inline doc/tutorials/trustyACRN.rst]: remove space: 
(  encrypt ....

File: doc/tutorials/trustyACRN.rst (hunk 2 above)
Comment (bzhu5) [reply doc/tutorials/trustyACRN.rst]: name could be "Keystore service" or just simply "keystore"

**Pull request merged**